File: UniqueNumberOfOccurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        # Time Complexity: O(n)
        # Space Complexity: O(n)
        freq = {}
        for i in arr:
            freq[i] = freq.get(i, 0) + 1
        uniques = set()
        for i in freq.values():
            if i in uniques:
                return False
            uniques.add(i)
        return True

# Approach:
# Store the elements and their corresponding frequencies in a hashmap.
# Iterate over the hashmap to check for repeating values.

# Time Complexity: O(n)
# Space Complexity: O(n)

File: UniqueNumberOfOccurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        // Time Complexity: O(n)
        // Space Complexity: O(n)
        HashMap<Integer, Integer> freq = new HashMap<>();
        for(int i: arr) {
            freq.put(i, freq.getOrDefault(i, 0)+1);
        }
        Set<Integer> uniques = new HashSet<>();
        for(int i: freq.values()) {
            if(uniques.contains(i)) {
                return false;
            }
            uniques.add(i);
        }
        return true;
    }
}

// Approach:
// Store the elements and their corresponding frequencies in a hashmap.
// Iterate over the hashmap to check for repeating values.

// Time Complexity: O(n)
// Space Complexity: O(n)
